### ISSUE TITLE
Training on rtx 2080 ti with cuda 9 is very slow.

### ISSUE THREAD

**Issue by jyothir07** [matterport/Mask_RCNN#1938]
Training on 2080 ti with Cuda 9.0 is very slow. Program is running but approximately 3-5% gpu utilization. memory is allocated. tensorflow-gpu==1.13.1, keras=2.1.3
Anyone solved this problem?? Thanks in advance
![post](https://user-images.githubusercontent.com/49629788/72407035-48835a00-3784-11ea-9a1d-8e17691718ad.png)

**Comment by marcfielding1** [comment]
Thats odd, err you've got the number of GPU's specified in your config and you're absolutely sure you've got tf-gpu rather than just tf?

**Comment by jyothir07** [comment]
> Thats odd, err you've got the number of GPU's specified in your config and you're absolutely sure you've got tf-gpu rather than just tf?

Yes. Absolutely sure. Installed tf-gpu 1.5.0. Device is recognised by tf. Memory allocated. GPU utilisation is still very low. thanks for ur reply.

**Comment by gaowq2017** [comment]
The result is not ture! You can use this software GPU-Z, the download link is here:https://www.techpowerup.com/download/gpu-z/

**Comment by jyothir07** [comment]
why is it not true? I checked it's still using GPU at about 4-5 %

**Comment by saskra** [comment]
I have the same problem, GPU memory is allocated but calculations take place on CPU instead of GPU.

**Comment by Osdel** [comment]
> Training on 2080 ti with Cuda 9.0 is very slow. Program is running but approximately 3-5% gpu utilization. memory is allocated. tensorflow-gpu==1.15.0, tensorflow-tensorboard==1.5.1,keras=2.1.3
> Anyone solved this problem?? Thanks in advance
> ![41813569-3bb40116-76ed-11e8-8b68-e05c33a187c7](https://user-images.githubusercontent.com/49629788/71620063-aee97380-2bed-11ea-8a54-cea92f4a45b3.png)

I am facing the same issue. Do you solve it?

**Comment by marcfielding1** [comment]
Thats just plain odd, do you have some code somewhere I can have a look at - the team might be better placed to support you but I have some knowledge that got me going along nicely. tf-gpu is number one, I'm also using Cuda 10:

Could you post your training startup log so where it finds the GPU and talks about loading lubcudalas etc.

**Comment by Osdel** [comment]
> Thats just plain odd, do you have some code somewhere I can have a look at - the team might be better placed to support you but I have some knowledge that got me going along nicely. tf-gpu is number one, I'm also using Cuda 10:
> 
> Could you post your training startup log so where it finds the GPU and talks about loading lubcudalas etc.

![image](https://user-images.githubusercontent.com/33522636/72302987-bc651b80-3639-11ea-929a-96ddcd40712b.png)

This is my startup log, it detect my Nvidia 1050 Ti properly. I think the problem is becuse memory allocation issues so it wastes time allocating and that create a bottleneck. But anyway 5%-10% is a very low utilization rate.

**Comment by jyothir07** [comment]
> > Thats just plain odd, do you have some code somewhere I can have a look at - the team might be better placed to support you but I have some knowledge that got me going along nicely. tf-gpu is number one, I'm also using Cuda 10:
> > Could you post your training startup log so where it finds the GPU and talks about loading lubcudalas etc.
> 
> ![image](https://user-images.githubusercontent.com/33522636/72302987-bc651b80-3639-11ea-929a-96ddcd40712b.png)
> 
>  This is my startup log, it detect my Nvidia 1050 Ti properly. I think the problem is becuse memory allocation issues so it wastes time allocating and that create a bottleneck. But anyway 5%-10% is a very low utilization rate.

 @osdel
(defect-detection) E:\jyo\gdxray_mod>python gdxray.py
2020-01-14 16:10:57.549042: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
C:\Users\User\Anaconda3\envs\defect-detection\lib\site-packages\h5py\__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
Command:  train
Model:  E:/jyo/gdxray_mod/mask_rcnn_coco.h5
Dataset:  E:/jyo/gdxray_mod/
Series:  All
Logs:  E:\jyo\gdxray_mod\logs/gdxray

Configurations:
BACKBONE_SHAPES                [[128 128]
 [ 64  64]
 [ 32  32]
 [ 16  16]
 [  8   8]]
BACKBONE_STRIDES               [4, 8, 16, 32, 64]
BATCH_SIZE                     1
BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]
DETECTION_MAX_INSTANCES        100
DETECTION_MIN_CONFIDENCE       0.7
DETECTION_NMS_THRESHOLD        0.3
GPU_COUNT                      1
IMAGES_PER_GPU                 1
IMAGE_MAX_DIM                  512
IMAGE_MIN_DIM                  512
IMAGE_PADDING                  True
IMAGE_SHAPE                    [512 512   3]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               100
MEAN_PIXEL                     [123.7 116.8 103.9]
MINI_MASK_SHAPE                (128, 128)
NAME                           gdxray
NUM_CLASSES                    3
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        1000
POST_NMS_ROIS_TRAINING         2000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.5, 1, 2]
RPN_ANCHOR_SCALES              (16, 32, 64, 128, 256)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]
RPN_NMS_THRESHOLD              0.5
RPN_TRAIN_ANCHORS_PER_IMAGE    256
STEPS_PER_EPOCH                100
TRAIN_ROIS_PER_IMAGE           50
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               50
WEIGHT_DECAY                   0.0001


2020-01-14 16:11:03.756841: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2
2020-01-14 16:11:03.765504: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library nvcuda.dll
2020-01-14 16:11:03.798277: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties:
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.545
pciBusID: 0000:01:00.0
2020-01-14 16:11:03.805896: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-01-14 16:11:03.812990: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-01-14 16:11:03.820884: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-01-14 16:11:03.825749: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-01-14 16:11:03.832678: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-01-14 16:11:03.840962: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-01-14 16:11:03.852580: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-01-14 16:11:03.857223: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-01-14 16:11:04.421007: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-01-14 16:11:04.424892: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0
2020-01-14 16:11:04.427936: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N
2020-01-14 16:11:04.431449: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5)
.
.
.
Adding images......
.
.

Training network heads

Starting at epoch 0. LR=0.001

Checkpoint Path: E:\jyo\gdxray_mod\logs/gdxray\gdxray20200114T1641\mask_rcnn_gdxray_{epoch:04d}.h5
Selecting layers to train
fpn_c5p5               (Conv2D)
fpn_c4p4               (Conv2D)
fpn_c3p3               (Conv2D)
fpn_c2p2               (Conv2D)
fpn_p5                 (Conv2D)
fpn_p2                 (Conv2D)
fpn_p3                 (Conv2D)
fpn_p4                 (Conv2D)
In model:  rpn_model
    rpn_conv_shared        (Conv2D)
    rpn_class_raw          (Conv2D)
    rpn_bbox_pred          (Conv2D)
mrcnn_mask_conv1       (TimeDistributed)
mrcnn_mask_bn1         (TimeDistributed)
mrcnn_mask_conv2       (TimeDistributed)
mrcnn_mask_bn2         (TimeDistributed)
mrcnn_class_conv1      (TimeDistributed)
mrcnn_class_bn1        (TimeDistributed)
mrcnn_mask_conv3       (TimeDistributed)
mrcnn_mask_bn3         (TimeDistributed)
mrcnn_class_conv2      (TimeDistributed)
mrcnn_class_bn2        (TimeDistributed)
mrcnn_mask_conv4       (TimeDistributed)
mrcnn_mask_bn4         (TimeDistributed)
mrcnn_bbox_fc          (TimeDistributed)
mrcnn_mask_deconv      (TimeDistributed)
mrcnn_class_logits     (TimeDistributed)
mrcnn_mask             (TimeDistributed)


C:\Users\User\Anaconda3\envs\defect-detection\lib\site-packages\tensorflow_core\python\framework\indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
C:\Users\User\Anaconda3\envs\defect-detection\lib\site-packages\tensorflow_core\python\framework\indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
C:\Users\User\Anaconda3\envs\defect-detection\lib\site-packages\tensorflow_core\python\framework\indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "


Epoch 1/40
2020-01-14 16:42:23.869616: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-01-14 16:42:24.145020: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-01-14 16:42:25.303111: W tensorflow/stream_executor/cuda/redzone_allocator.cc:312] Internal: Invoking ptxas not supported on Windows
Relying on driver to perform ptx compilation. This message will be only logged once.
  1/100 [..............................] - ETA: 16:06 - loss: 7.0612 - rpn_class_loss: 0.0176 - rpn_bbox_loss: 2.5934 - mrcnn_class_loss: 4.4502 - mrcnn_bbox_loss: 0.0000e+00 - mrcnn_mask_loss: 0.0000e+00

**Comment by codeme-09** [comment]
> > Training on 2080 ti with Cuda 9.0 is very slow. Program is running but approximately 3-5% gpu utilization. memory is allocated. tensorflow-gpu==1.15.0, tensorflow-tensorboard==1.5.1,keras=2.1.3
> > Anyone solved this problem?? Thanks in advance
> > ![41813569-3bb40116-76ed-11e8-8b68-e05c33a187c7](https://user-images.githubusercontent.com/49629788/71620063-aee97380-2bed-11ea-8a54-cea92f4a45b3.png)
> 
> I am facing the same issue. Do you solve it?

Nope... Still trying

**Comment by Chakib-Belafdil** [comment]
Which python version are you using ?

**Comment by jyothir07** [comment]
@Chakib-Belafdil  everything was created with python 3.6 environment

**Comment by jyothir07** [comment]
> The result is not ture! You can use this software GPU-Z, the download link is here:https://www.techpowerup.com/download/gpu-z/

its true and confirmed with gpu-z

**Comment by jyothir07** [comment]
@Chakib-Belafdil, @Osdel, @marcfielding1 
I have tried different configurations but still, I haven't found a solution.
tf-gpu=1.14.0, keras=2.0.0, python=3.6, cuda=10, cudnn=7.6.5
tf-gpu=1.14.0, keras=2.0.0, python=**3.7**, cuda=10, cudnn=7.6.5
tf-gpu=1.15.0, keras=2.1.3, python=3.6, cuda=10, cudnn=7.6.5
tf-gpu=1.5.0, keras=2.0.8, python=3.6, cuda=9, cudnn=7.0.5

any help is appreciated.

**Comment by Chakib-Belafdil** [comment]
> @Chakib-Belafdil, @Osdel, @marcfielding1
> I have tried different configurations but still, I haven't found a solution.
> tf-gpu=1.14.0, keras=2.0.0, python=3.6, cuda=10, cudnn=7.6.5
> tf-gpu=1.14.0, keras=2.0.0, python=**3.7**, cuda=10, cudnn=7.6.5
> tf-gpu=1.15.0, keras=2.1.3, python=3.6, cuda=10, cudnn=7.6.5
> tf-gpu=1.5.0, keras=2.0.8, python=3.6, cuda=9, cudnn=7.0.5
> 
> any help is appreciated.

I will try the solution in this link this afternoon: https://medium.com/@salixabb/how-to-set-up-tensorflow-for-gpu-and-cpu-on-windows-10-recommended-on-virtual-environment-4fd937944674 

I will come back to you if it works

**Comment by Chakib-Belafdil** [comment]
@codeme-09 

Try setting in NVIDIA control panel the 3D to maximum performance as said in last post here :

https://stackoverflow.com/questions/58867071/keras-rtx-2080-ti-training-slower-than-both-cpu-only-and-gtx-1070 

My issue is still on, I can't even make it work on slow mode :(

**Comment by Chakib-Belafdil** [comment]
Check here. I made it work on my RTX 2080 Ti (Windows 10):
https://github.com/matterport/Mask_RCNN/issues/1952

**Comment by jyothir07** [comment]
@Chakib-Belafdil  will try and update. thanks bro

**Comment by jyothir07** [comment]
@Chakib-Belafdil thank you so much for your time for preparing all this and providing the right guidelines. I followed your instructions and reduced epochs. now the Average GPU utilisation is upto 9%. even though I couldn't improve so much I'm happy with atleast this much improvement. once again I appreciate your  work..

**Comment by shining-love** [comment]
> I have the same problem, GPU memory is allocated but calculations take place on CPU instead of GPU.

I have the same problem,how to fix it?

**Comment by jyothir07** [comment]
> > I have the same problem, GPU memory is allocated but calculations take place on CPU instead of GPU.
> 
> I have the same problem,how to fix it?

Haven't fixed it completely. Just follow the above procedure. Avg Utilization in the range of 20%

**Comment by burhr2** [comment]
Hi! kindly see this issue https://github.com/matterport/Mask_RCNN/issues/2312 there is a pointer to the same implementation that supports TensorFlow 2=> (with GPU). If this solves your problem kindly close the issue so as others can navigate to other issues easier

**Comment by Xi-Gong** [comment]
2080Ti with Pascal architecture support from CUDA 10.0, the official cuda toolkit doc mention it. Using older cuda with no support may result in unprecedented situation.
